sympy/ntheory/ecm.py

```diff
@@ -0,0 +1,244 @@
+        return n
+
+    curve = 0
+    D = int(sqrt(B2))
+    beta = [0]*(D + 1)
+    S = [0]*(D + 1)
+    k = 1
+    for p in sieve.primerange(1, B1 + 1):
+        k *= pow(p, integer_log(B1, p)[0])
+    g = 1
+
+    random.seed(seed)
+    while(curve <= max_curve):
+        curve += 1
+
+        #Suyama's Paramatrization
+        sigma = random.randint(6, n)
+        u = (sigma*sigma - 5) % n
+        v = (4*sigma) % n
+        diff = v - u
+        u_3 = pow(u, 3, n)
+
+        try:
+            C = (pow(diff, 3, n)*(3*u + v)*mod_inverse(4*u_3*v, n) - 2) % n
+        except ValueError:
+            #If the mod_inverse(4*u_3*v, n) doesn't exist
+            return gcd(4*u_3*v, n)
+
+        a24 = (C + 2)*mod_inverse(4, n) % n
+        Q = Point(u_3 , pow(v, 3, n), a24, n)
+        Q = Q.mont_ladder(k)
+        g = gcd(Q.z_corr, n)
+
+        #Stage 1 factor
+        if g != 1 and g != n:
+            return g
+        #Stage 1 failure. Q.z = 0, Try another curve
+        elif g == n:
+            continue
+
+        #Stage 2 - Improved Standard Continuation
+        S[1] = Q.double()
+        S[2] = S[1].double()
+        beta[1] = (S[1].x_corr*S[1].z_corr) % n
+        beta[2] = (S[2].x_corr*S[2].z_corr) % n
+
+        for d in range(3, D + 1):
+            S[d] = S[d - 1].add(S[1], S[d - 2])
+            beta[d] = (S[d].x_corr*S[d].z_corr) % n
+
+        g = 1
+        B = B1 - 1
+        T = Q.mont_ladder(B - 2*D)
+        R = Q.mont_ladder(B)
+
+        for r in  range(B, B2, 2*D):
+            alpha = (R.x_corr*R.z_corr) % n
+            for q in sieve.primerange(r + 2, r + 2*D + 1):
+                delta = (q - r) // 2
+                f = (R.x_corr - S[d].x_corr)*(R.z_corr + S[d].z_corr) - alpha + beta[delta]
+                g = (g*f) % n
+            #Swap
+            TT = R
+            R = R.add(S[D], T)
+            T = TT
+        g = gcd(n, g)
+
+        #Stage 2 Factor found
+        if g != 1 and g != n:
+            return g
+
+    #ECM failed, Increase the bounds
+    raise ValueError("Increase the bounds")
+
+
+def ecm(n, B1=10000, B2=100000, max_curve=200, increase_bound=False, seed=1234):
+    """Performs factorization using
+    Lenstra's Elliptic curve method.
+
+    Parameters:
+    ===========
+
+    n : Number to be Factored
+    B1 : Stage 1 Bound
+    B2 : Stage 2 Bound
+    max_curve : Maximum number of curves generated
+    increase_bound : If True, then the Stage 1 and 2 bounds increase
+        by a factor of 10 incase of ECM failure
+    seed : Initialize pseudorandom generator
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory import ecm
+    >>> ecm(25645121643901801)
+    {5394769, 4753701529}
+    >>> ecm(9804659461513846513)
+    {4641991, 2112166839943}
+    """
+    factor = set()
+    for i in sieve.primerange(1, 100000):
+        if n % i == 0:
+            factor.add(i)
+            while(n % i == 0):
+                n //= i
+    while(n % 2 == 0):
+        n //= 2
+        factor.add(2)
```

Comment:
I m bit confused. why are you checking divisibility of 2 as I think it will already be checked in the above for loop.